.github/release_pull_request_template.md

```diff
@@ -0,0 +1,42 @@
+This is the release pull request for {version} of the WooCommerce Blocks plugin.
+
+## Communication
+
+<!--
+  This section is for any notes related to communicating the release.
+  Please include any extra details with each item as needed.
+-->
+
+* [ ] What is being introduced in this release?
+* [ ]  If this release has potentially **high customer impact**, is a blog post (section) ready? Provide a link if applicable.
+* [ ] Is the **relevant developer documentation updated**? Please provide links if applicable.
+* [ ] Any special instructions or helpful notes for Happiness engineers are posted (this might not be needed for every release).
+
+## Quality
+
+<!--
+  This section is for any notes related to quality around the release.
+  Please include any extra details with each item as needed.
+-->
+
+* [ ] Have you **written tests** to cover the changes?
+     * [ ] Unit tests
+     * [ ] E2E tests
+     * [ ] for each supported WordPress and WooCommerce core versions.
+* [ ] Have you tested on the applicable platforms?
+     * [ ] mobile
+     * [ ] desktop
+* [ ] Does this release affect API's and conform to API versioning policy?
+* [ ] Have you considered the impact of the changes in the release on **other extensions** and **backward compatibility**?
+     * [ ] Does the release change the signature of any public methods or functions?
+     * [ ] Are filters or hooks affected by the PR code changes?
+* [ ] Does the release include a changelog entry in the readme.txt?
+* [ ] Please provide a link to **testing instructions** for this release.
+* [ ] Have you reviewed impacts to performance (bundle sizes, query time etc)?
```

Comment:
How are we meant to review this at this stage? It's done per PR during the cycle.

Reply:
There's two aspects to this:

- First, yes each pull has (or should have) a bundle size (performance impact) measurement. But sometimes given the development that happens over a longer period of time (or when a feature behind a gate has that gate removed for it's initial release) the impact on bundle size is needing more obvious call-out/decision. An example is 2.6.0 where we noticed that All Products and Filter blocks bundle sizes jumped significantly, but we decided to release anyways (with an eye to figuring out why the jump and improving in later iteration):

> Note: As noted by @senadir here, there is some impact to bundle sizes for frontend js built for the All Products and Filter blocks. We are aware of this impact which is caused by some dependencies pulling in DashIcons and have a potential fix that is not ready for this release. The target is 2.7.0 for having this fixed so that when the feature plugin is updated in WooCommerce core this impact is not included there.

(Note: see #2280 - which I just realized wasn't prioritized that will address this, so updated)

- Second, this is another place where the release lead _should_ review the changeset in the release and test any areas where there may be concern around performance degradation (that might not have been caught in individual pulls) or raise the flag for feedback on suspected degradations so that the team can weigh in on whether it's of concern or not and why. **Yes** this potentially slows down a release, but in most cases this would simply be a matter of checking the box because of our current process of doing those checks with each pull.

One thing I can do here is add a comment block for folks to add any notes about this if needed.

Reply:
In the changes I'm pushing shortly, I thought it'd be good to break this out into known _negative_ performance impacts and known _positive_ performance impacts (because it's just as important to highlight positive gains along with negative gains!). 

Sometimes this info will be included in release posts, sometimes not, so I think it's still useful to break out into it's own section as a part of the "quality" area.